### PR TITLE
qt-svg: new package for Qt6 SVG module

### PR DESCRIPTION
**Pull request by aumuell**
Depends on #40833 

enables loading of SVG icons by providing plugin used by qt-base

### CODE REVIEW

File: var/spack/repos/builtin/packages/qt-svg/package.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+from spack.package import *
+from spack.pkg.builtin.qt_base import QtBase, QtPackage
+
+
+class QtSvg(QtPackage):
+    """Scalable Vector Graphics (SVG) is an XML-based language for describing
+    two-dimensional vector graphics. Qt provides classes for rendering and
+    displaying SVG drawings in widgets and on other paint devices."""
+
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/qt-svg/package.py:15]: I realize there is a default `qt` homepage but is it worth adding a more specific one here (i.e., `https://doc.qt.io/qt-6/qtsvg-index.html`)?

File: var/spack/repos/builtin/packages/qt-svg/package.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    url = QtPackage.get_url(__qualname__)
+    list_url = QtPackage.get_list_url(__qualname__)
+
+    version("6.6.0", sha256="4fd6b4d9307c3cd8fd207e60334823fed07a9acb32f7d53cd9c9be9b6a2f8a30")
+    version("6.5.3", sha256="fb8e5574c2480aab78062fad2d0a521633b4591ada600130b918b703c2ddb09a")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/qt-svg/package.py:20]: These versions aren't yet made available in the `qt-base` package so the concretizer will resolve that dependency to v6.5.2.  Is that desirable?  If not, you probably want to update the `qt-base` package as well or remove these versions for now.

**Review by tldahlgren — CHANGES_REQUESTED** [review]
I have confirmed the 10 version sha256 but have questions on this PR.

File: var/spack/repos/builtin/packages/qt-svg/package.py (hunk 2 above)
Comment (tldahlgren) [reply var/spack/repos/builtin/packages/qt-svg/package.py:20]: Nvm.  I just saw the PR to update those so I'll add a dependencies in the description.

**Review by tldahlgren — APPROVED** [review]
I still have the question about the homepage though.

**Comment by tldahlgren** [comment]
@sethrj @wdconinc Since you both appear to be maintainers on a number of `qt` packages, do you want to comment on this one?

**Comment by sethrj** [comment]
I'll defer to @wdconinc ; he's the main qt user these days, and now he has magical merge privileges! 🎉

File: var/spack/repos/builtin/packages/qt-svg/package.py (hunk 1 above)
Comment (aumuell) [reply var/spack/repos/builtin/packages/qt-svg/package.py:15]: This probably makes sense, but not just for qt-svg: all the Qt packages except for qt-base and qt-declarative could do so. Here is a draft for changing that: #40850

**Review by wdconinc — APPROVED** [review]
Both most recent version tested, and build fine on my test system (without widgets variant).

**Comment by wdconinc** [comment]
> he's the main qt user these days

I keep waiting for geant4 to support Qt6...

**Pull request merged**